### PR TITLE
shape for all models verified

### PR DESCRIPTION
**Pull request by ckolbPTB**
Wasabi and Wasabiti now also compatible with new signal model dimensions.

I need help with the variable names here: https://github.com/PTB-MR/mrpro/blob/f8b056248c6705817e0daf81f37eda3199d10511/tests/operators/models/test_shape_all_models.py#L13

The first is the model name (easy), the second is the number of input tensors of `forward` and the third number is the number of tensors required for the `init`. Any good suggestions of how to name them?

### CODE REVIEW

**Comment by ckolbPTB** [comment]
closes #227 
closes #202

File: src/mrpro/operators/models/_WASABITI.py [hunk 1]
```diff
@@ -72,21 +76,24 @@ def forward(self, b0_shift: torch.Tensor, rb1: torch.Tensor, t1: torch.Tensor) -
...
-        # ensure correct dimensionality
-        offsets = self.offsets[(...,) + (None,) * b0_shift.ndim]
-        trec = self.trec[(...,) + (None,) * b0_shift.ndim]
+        delta_ndim = b0_shift.ndim - (self.offsets.ndim - 1)  # -1 for offset
+        offsets = self.offsets[..., *[None] * (delta_ndim)] if delta_ndim > 0 else self.offsets
+        trec = self.trec[..., *[None] * (delta_ndim)] if delta_ndim > 0 else self.trec
 
+        b1 = self.b1_nom * rb1
         da = offsets - b0_shift
         mz_initial = 1.0 - torch.exp(torch.multiply(-1.0 / t1, trec))
```
Comment (fzimmermann89) [inline src/mrpro/operators/models/_WASABITI.py]: Why torch.multiply vs torch.exp(-trec / t1)

Comment (fzimmermann89) [reply src/mrpro/operators/models/_WASABITI.py:98]: Why torch.multiply vs torch.exp(-trec / t1)

**Review by fzimmermann89 — APPROVED** [review]

**Comment by fzimmermann89** [comment]
num_init_parameters -> n_init_parameters ?
and maybe mention this as well in comment you linked 
`# Signal model, number of tensor arguments in forward (i.e. quantitative parameters) , number of tensor arguments in init (e.g. inversion time or CEST offsets)`
or similar?

**Pull request merged**